Build failure: pdfmixtool

### Steps To Reproduce

Steps to reproduce the behavior:
1. build pdfmixtool on unstable

### Build log

```
nix log /nix/store/5wfh4n7gqfb6f668ipp8q99pfp4g3d7k-pdfmixtool-1.1.1.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/5wfh4n7gqfb6f668ipp8q99pfp4g3d7k-pdfmixtool-1.1.1.drv^*'
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/q9x867gh9sxxfmzv7n583y24x95vxvqs-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
applying patch /nix/store/1mhx22d0ibg59av6z2mz8ayngan5y0fm-bd5f78c3a4d977d9b0c74302ce2521c737189b43.diff
patching file CMakeLists.txt
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGIS>
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/khkhbch4p1wjfl1g89gw1mszvvr7bzv0-gcc-wrapper-13.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/khkhbch4p1wjfl1g89gw1mszvvr7bzv0-gcc-wrapper-13.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /nix/store/01nygk9550khsba4msxd751bvgs8jj23-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2>
-- Checking for module 'libqpdf'
--   Found libqpdf, version 11.6.3
-- Checking for module 'Magick++'
--   Found Magick++, version 7.1.1
-- Configuring done (1.8s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j12 SHELL=/nix/store/cjbyb45nxiqidj95c4k1mh65azn1x896-bash-5.2-p21/bin/bash
[  1%] Automatic MOC for target pdfmixtool
[  1%] Built target pdfmixtool_autogen
[  3%] Generating pdfmixtool_zh_Hans.qm
[  5%] Generating pdfmixtool_ja.qm
[  6%] Automatic RCC for resources/resources.qrc
[  8%] Generating pdfmixtool_tr.qm
[ 10%] Generating pdfmixtool_it.qm
[ 12%] Generating pdfmixtool_hu.qm
[ 13%] Generating pdfmixtool_pt.qm
[ 15%] Generating pdfmixtool_pt_BR.qm
[ 17%] Generating pdfmixtool_nb_NO.qm
[ 18%] Generating pdfmixtool_ru.qm
[ 20%] Generating pdfmixtool_id.qm
[ 22%] Generating pdfmixtool_de.qm
Updating '/build/source/build/pdfmixtool_pt.qm'...
Updating '/build/source/build/pdfmixtool_pt_BR.qm'...
Updating '/build/source/build/pdfmixtool_hu.qm'...
Updating '/build/source/build/pdfmixtool_tr.qm'...
Updating '/build/source/build/pdfmixtool_ru.qm'...
Updating '/build/source/build/pdfmixtool_it.qm'...
Updating '/build/source/build/pdfmixtool_zh_Hans.qm'...
Updating '/build/source/build/pdfmixtool_ja.qm'...
Updating '/build/source/build/pdfmixtool_id.qm'...
Updating '/build/source/build/pdfmixtool_nb_NO.qm'...
    Generated 167 translation(s) (167 finished and 0 unfinished)
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
    Generated 167 translation(s) (167 finished and 0 unfinished)
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
    Generated 141 translation(s) (141 finished and 0 unfinished)
    Ignored 26 untranslated source text(s)
    Generated 167 translation(s) (167 finished and 0 unfinished)
    Generated 142 translation(s) (102 finished and 40 unfinished)
    Ignored 25 untranslated source text(s)
    Generated 167 translation(s) (167 finished and 0 unfinished)
    Generated 144 translation(s) (102 finished and 42 unfinished)
    Ignored 23 untranslated source text(s)
    Generated 161 translation(s) (161 finished and 0 unfinished)
    Ignored 6 untranslated source text(s)
    Generated 167 translation(s) (120 finished and 47 unfinished)
    Generated 167 translation(s) (167 finished and 0 unfinished)
Updating '/build/source/build/pdfmixtool_de.qm'...
    Generated 167 translation(s) (167 finished and 0 unfinished)
[ 24%] Generating pdfmixtool_hr.qm
[ 29%] Generating pdfmixtool_he.qm
[ 29%] Generating pdfmixtool_fr.qm
[ 29%] Generating pdfmixtool_si.qm
[ 31%] Generating pdfmixtool_sv.qm
[ 32%] Generating pdfmixtool_cs.qm
[ 34%] Generating pdfmixtool_es.qm
[ 36%] Generating pdfmixtool_ca.qm
[ 39%] Generating pdfmixtool_nl.qm
[ 39%] Generating pdfmixtool_pl.qm
[ 41%] Generating pdfmixtool_en.qm
Updating '/build/source/build/pdfmixtool_hr.qm'...
    Generated 167 translation(s) (167 finished and 0 unfinished)
Updating '/build/source/build/pdfmixtool_he.qm'...
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
    Generated 167 translation(s) (167 finished and 0 unfinished)
Updating '/build/source/build/pdfmixtool_fr.qm'...
Updating '/build/source/build/pdfmixtool_si.qm'...
    Generated 3 translation(s) (2 finished and 1 unfinished)
    Ignored 164 untranslated source text(s)
Updating '/build/source/build/pdfmixtool_sv.qm'...
Updating '/build/source/build/pdfmixtool_pl.qm'...
    Generated 167 translation(s) (164 finished and 3 unfinished)
    Generated 6 translation(s) (6 finished and 0 unfinished)
    Ignored 161 untranslated source text(s)
    Generated 167 translation(s) (167 finished and 0 unfinished)
Updating '/build/source/build/pdfmixtool_cs.qm'...
Updating '/build/source/build/pdfmixtool_es.qm'...
    Generated 80 translation(s) (52 finished and 28 unfinished)
    Ignored 87 untranslated source text(s)
    Generated 167 translation(s) (167 finished and 0 unfinished)
Updating '/build/source/build/pdfmixtool_ca.qm'...
Updating '/build/source/build/pdfmixtool_nl.qm'...
    Generated 167 translation(s) (167 finished and 0 unfinished)
    Generated 167 translation(s) (167 finished and 0 unfinished)
Updating '/build/source/build/pdfmixtool_en.qm'...
    Generated 1 translation(s) (1 finished and 0 unfinished)
    Ignored 166 untranslated source text(s)
[ 43%] Automatic RCC for tr.qrc
[ 48%] Building CXX object CMakeFiles/pdfmixtool.dir/src/main.cpp.o
[ 50%] Building CXX object CMakeFiles/pdfmixtool.dir/src/aboutdialog.cpp.o
[ 50%] Building CXX object CMakeFiles/pdfmixtool.dir/src/editmultipageprofiledialog.cpp.o
[ 50%] Building CXX object CMakeFiles/pdfmixtool.dir/pdfmixtool_autogen/mocs_compilation.cpp.o
[ 51%] Building CXX object CMakeFiles/pdfmixtool.dir/src/editpdfentrydialog.cpp.o
[ 53%] Building CXX object CMakeFiles/pdfmixtool.dir/src/gui_utils.cpp.o
[ 55%] Building CXX object CMakeFiles/pdfmixtool.dir/src/inputpdffiledelegate.cpp.o
[ 58%] Building CXX object CMakeFiles/pdfmixtool.dir/src/inputpdffilewidget.cpp.o
[ 58%] Building CXX object CMakeFiles/pdfmixtool.dir/src/mainwindow.cpp.o
[ 60%] Building CXX object CMakeFiles/pdfmixtool.dir/src/mouseeventfilter.cpp.o
[ 62%] Building CXX object CMakeFiles/pdfmixtool.dir/src/multipageprofilesmanager.cpp.o
[ 63%] Building CXX object CMakeFiles/pdfmixtool.dir/src/pdf_edit_lib/definitions.cpp.o
In file included from /nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/Magick++/Inclu>
                 from /nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/Magick++.h:12,
                 from /build/source/src/main.cpp:22:
/nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/MagickCore/magick-config.h:63:3: war>
   63 | # warning "you should set MAGICKCORE_QUANTUM_DEPTH to sensible default set it to configure time default"
      |   ^~~~~~~
/nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/MagickCore/magick-config.h:63:3: war>
/nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/MagickCore/magick-config.h:64:3: war>
   64 | # warning "this is an obsolete behavior please fix your makefile"
      |   ^~~~~~~
/nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/MagickCore/magick-config.h:64:3: war>
/nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/MagickCore/magick-config.h:86:3: err>
   86 | # error "you should set MAGICKCORE_HDRI_ENABLE"
      |   ^~~~~
/nix/store/db0a91hspgbq5lsjaf5q352jw0477k6w-imagemagick-7.1.1-27-dev/include/ImageMagick-7/MagickCore/magick-config.h:121:3: er>
  121 | # error "you should set MAGICKCORE_HDRI_ENABLE"
      |   ^~~~~
[ 65%] Building CXX object CMakeFiles/pdfmixtool.dir/src/pdf_edit_lib/pdf_editor.cpp.o
[ 67%] Building CXX object CMakeFiles/pdfmixtool.dir/src/pdf_edit_lib/pdf_info.cpp.o
[ 68%] Building CXX object CMakeFiles/pdfmixtool.dir/src/operations/abstract_operation.cpp.o
[ 70%] Building CXX object CMakeFiles/pdfmixtool.dir/src/operations/add_empty_pages.cpp.o
[ 72%] Building CXX object CMakeFiles/pdfmixtool.dir/src/operations/alternate_mix.cpp.o
[ 74%] Building CXX object CMakeFiles/pdfmixtool.dir/src/operations/booklet.cpp.o
make[2]: *** [CMakeFiles/pdfmixtool.dir/build.make:226: CMakeFiles/pdfmixtool.dir/src/main.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/build/source/src/pdf_edit_lib/pdf_editor.cpp: In member function 'void PdfEditor::add_pages(unsigned int, int, const PageLayou>
/build/source/src/pdf_edit_lib/pdf_editor.cpp:269:13: warning: 'parent_outline_index' may be used uninitialized [-Wmaybe-uninit>
  269 |         if (parent_outline_index == m_output_outlines.size() - 1)
      |             ^~~~~~~~~~~~~~~~~~~~
/build/source/src/pdf_edit_lib/pdf_editor.cpp:114:18: note: 'parent_outline_index' was declared here
  114 |     unsigned int parent_outline_index;
      |                  ^~~~~~~~~~~~~~~~~~~~
/build/source/src/operations/add_empty_pages.cpp: In member function 'void AddEmptyPages::m_save()':
/build/source/src/operations/add_empty_pages.cpp:202:16: warning: 'page_width' may be used uninitialized [-Wmaybe-uninitialized]
  202 |     page_width = page_width * cm;
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~
/build/source/src/operations/add_empty_pages.cpp:168:12: note: 'page_width' was declared here
  168 |     double page_width, page_height;
      |            ^~~~~~~~~~
/build/source/src/operations/add_empty_pages.cpp:203:17: warning: 'page_height' may be used uninitialized [-Wmaybe-uninitialize>
  203 |     page_height = page_height * cm;
      |     ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/build/source/src/operations/add_empty_pages.cpp:168:24: note: 'page_height' was declared here
  168 |     double page_width, page_height;
      |                        ^~~~~~~~~~~
make[1]: *** [CMakeFiles/Makefile2:84: CMakeFiles/pdfmixtool.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

Has not been been build by hydra since 09.2023
Latest error: https://hydra.nixos.org/build/247992921

### Notify maintainers

@onny 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
└─>🧙  nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/crx575dqd0kpvqmimh623lwsm35fqsy8-nix-info
copying path '/nix/store/crx575dqd0kpvqmimh623lwsm35fqsy8-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.75, NixOS, 24.05 (Uakari), 24.05.20240129.97b17f3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.3`
 - channels(root): `"nixos"`
 - channels(leon): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/office/pdfmixtool/default.nix
+++ b/pkgs/applications/office/pdfmixtool/default.nix
@@ -2,6 +2,7 @@
 , mkDerivation
 , fetchFromGitLab
 , fetchpatch
+, fetchpatch2
 , cmake
 , pkg-config
 , qtbase
@@ -40,6 +41,11 @@ mkDerivation rec {
     (fetchpatch {
       url = "https://gitlab.com/scarpetta/pdfmixtool/-/commit/bd5f78c3a4d977d9b0c74302ce2521c737189b43.diff";
       hash = "sha256-h2g5toFqgEEnObd2TYQms1a1WFTgN7VsIHyy0Uyq4/I=";
+    })
+    # https://gitlab.com/scarpetta/pdfmixtool/-/merge_requests/14
+    (fetchpatch2 {
+      url = "https://gitlab.com/scarpetta/pdfmixtool/-/commit/268291317ccd1805dc1c801ff88641ba06c6a7f0.patch";
+      hash = "sha256-56bDoFtE+IOQHcV9xPfyrgYYFvTfB0QiLIfRl91irb0=";
     })
   ];
 

